src/error.ts

```diff
@@ -0,0 +1,103 @@
+import { inspect } from 'node:util'
+import { Schema, YAMLMap, CollectionTag } from 'yaml'
+import { CreateNodeContext, ToJSContext } from 'yaml/util'
+
+const tag = '!error'
+
+// If the user cared to provide a custom inspect, then use
+// that as the source of extra properties.
+const customInspectSymbol: symbol = inspect.custom
+
+class YAMLErrorObject extends YAMLMap {
+  tag = tag
+  toJSON(_: unknown, ctx: ToJSContext) {
+    ctx = { ...ctx, mapAsMap: false }
+    const { name, message, ...rest } = super.toJSON(_, ctx)
+    const Cls =
+      name === 'EvalError'
+        ? EvalError
+        : name === 'RangeError'
+        ? RangeError
+        : name === 'ReferenceError'
+        ? ReferenceError
+        : name === 'SyntaxError'
+        ? SyntaxError
+        : name === 'TypeError'
+        ? TypeError
+        : name === 'URIError'
+        ? URIError
+        : Error
+    const er = new Cls(message)
+    if (Cls.name !== name) {
+      Object.defineProperty(er, 'name', {
+        value: name,
+        enumerable: false,
+        configurable: true
+      })
+    }
+    if (rest.stack === undefined) er.stack = undefined
```

Comment:
This will never be true because `undefined` values are left out unless `keepUndefined` is set. And if it's set, they end up as `null` instead. This probably requires separate handling in the `from()` as well.

Reply:
It'd be undefined if it wasn't in the list. In which case, we want the resulting Error object to have a stack explicitly set to undefined, not to some weird stack pointing into this module where it was created.

Eg:

```yaml
!error
name: RangeError
message: hello
```

In that case, you want to get an error object that doesn't have a stack, since it wasn't set in the yaml data. But if you don't set it, then it'll have a stack that is misleading and weird.

Reply:
Ah, got it. Could the check then be something like this, to allow for a YAML serialization that's explicit about defining an empty stack?
```suggestion
    if (rest.stack == undefined) er.stack = undefined
```

Reply:
Seems fine to me. I usually do `x === undefined || x === null` to avoid `==` entirely, but it's your back yard :)

Reply:
Yeah, that looks pretty good. I'm rather glad that JS doesn't have too many magic properties like this.